### ISSUE TITLE
Provide a DB setting to accept only paths within the TMSU root

### ISSUE THREAD

**Issue by ipkiss42** [oniony/TMSU#207]
As a user, I would like my folders with a TMSU database to be relocatable, i.e. I don't want to store in the DB a file which "escapes" the root folder of the TMSU database. Since TMSU already stores relative paths whenever possible, we are almost there, but I can still tag an absolute path by mistake, or tag a symlink pointing outside of the root, or even be affected by #168.

It would be nice if TMSU allowed me to avoid mistakes by rejecting explicitly paths outside of the root, with a setting in the DB.

This setting might even become the default for new databases. On a typical system, I suspect that most databases would be "relative-only", except one, used as a fallback.

**Comment by oniony** [comment]
No, it's not always possible or desirable to have the database collocated
with the files. Think read-only media, network files, stuff on slow access
tape drives &c.

**Comment by ipkiss42** [comment]
OK, then it shouldn't be the default indeed.
